ls-ui-react-app/src/components/LatticeView.tsx

```diff
@@ -132,57 +133,90 @@ type LatticeViewProps = {
 }
 const LatticeView = ({compilationResult} : LatticeViewProps) => {
     const [selectedSliceNumber, setSelectedSliceNumber] = React.useState<number>(0);
-    const changeSlice = (delta: number) => setSelectedSliceNumber(selectedSliceNumber+delta)
-    const {compilationText, slices} = compilationResult
-    
+    const changeSlice = (delta: number) => {
+        setSelectedSliceNumber(selectedSliceNumber+delta)
+        scrollToLattice()
+    }
+    const {compilation_text, slices} = compilationResult
     const slices_len = slices.length
     // JS object that returns boolean when "previous" or "next" buttons need to be disabled
     const disable = {
         "prev": selectedSliceNumber===0,
         "next": selectedSliceNumber===slices_len-1 
     }
 
-    const latticeRef = useRef<HTMLInputElement>(null);
+    // scroll into Lattice View Section, one time, after compilation is completed
+    const latticeSection = useRef<HTMLInputElement>(null);
     React.useEffect(() => {
-        latticeRef.current && latticeRef!.current!.scrollIntoView();
-    })
+        latticeSection.current && latticeSection!.current!.scrollIntoView();
+    },[])
+
+    // scroll into Lattice View, after slice navigation buttons pressed
+    const lattice = useRef<HTMLInputElement>(null);
+    const scrollToLattice = () => {
+        lattice.current && lattice!.current!.scrollIntoView();
+    }
```

Comment:
I think there is some removable repetition here. There are `const latticeSection = useRef<HTMLInputElement>(null);` and `const lattice = useRef<HTMLInputElement>(null);`. I think you only need one of them.

Also,
```
React.useEffect(() => {
        latticeSection.current && latticeSection!.current!.scrollIntoView();
},[])
```
could just be `React.useEffect(scrollToLattice, [])`

Reply:
There are actually two different scroll into view events, one for when the compilation loads, and then a second one when the user changes slices. The reason for the second scroll event is because the compilation text takes up so much space on screen, that if it is toggled, the lattice view probably isn't completely visible. So if you press increment slice, the user is presumably done with viewing the compilation and wants to view the lattice, so it re-centers a second time. I can remove any subsequent re-centering by feeding an empty array to useEffect as indicated above.

Let's try out the auto-scrolling as is for a bit, and I can change if people find it annoying. I think its a good way to maximize the actual lattice view, which goes hand in hand with responsive rendering to fill any screen size

Reply:
Ok I see the difference, it's probably good then. Maybe naming these variables a little more different?

Reply:
The only thing I can suggest is changing `latticeSelection` to `sliceSelection`

Reply:
included more descriptive variable names